### ISSUE TITLE
[Bug]: child_process stderr & stdout pipe events not firing when executing a pre-compiled binary via electron on Windows

### ISSUE THREAD

**Issue by stephendpmurphy** [electron/electron#28492]
### Preflight Checklist

I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/master/CONTRIBUTING.md) for this project.
I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a feature request that matches the one I want to file, without success.

### Electron Version

12.0.2

### What operating system are you using?

Windows

### Operating System Version

Windows 10 - build 19042

### What arch are you using?

x64

### Expected Behavior

When executing a pre-compiled binary using the child_process.execFile OR child_process.spawn module, I expect the stderr and stdout pipes to have data events emitted which I can then use to log the output of my binary execution.

### Actual Behavior

When executing the script below using electron, no events are emitted via the stdout and stderr pipes. If I run the same script in Node, I get the proper output which shows individual lines of data being sent out via the stderr and stdout pipes.
```js
var promiseFromChildProcess = function(child) {
    return new Promise(function (resolve, reject) {
        child.addListener("error", reject);
        child.addListener("exit", resolve);
    });
}

var writeDeviceImage = async function(filePath) {
    // sam-ba -p serial -d sama5d2 -a bootconfig -c readcfg:bscr
    const appRootDir = require('app-root-dir').get();
    const samba = appRootDir + '/src/bin/sam-ba_3.3.1/sam-ba.exe'

    var exec = require('child_process').spawn;
    var param = ["-pserial", "-dsama5d2", "-asdmmc:1:1:0:0:4", `-cwrite:${filePath}`];

    child = exec(samba, param);

    // child.stderr.setEncoding('utf8');
    child.stderr.on('data', (data) => {
        console.log(data.toString());
    })

    console.log(`Writing ${filePath}`);

    await promiseFromChildProcess(child);

    console.log(`Complete: ${child.exitCode}`);

    return child.exitCode;
}

writeDeviceImage('test.img');
```

Output when executed via Electron
```bash
$ npx electron sam-ba.js

Writing test.img
Complete: 0
```

Output when executed via Node v15.5.0 - (Output shortened to keep issue somewhat brief)
```bash
$ node sam-ba.js

Writing test.img
Opening serial port 'COM24'

Connection opened.

Detected memory size is 15931539456 bytes.
Page size is 512 bytes.
Buffer is 88064 bytes (172 pages) at address 0x0022a620.

Executing command 'write:test.img'

Wrote 88064 bytes at address 0x00000000 (0.13%)
Wrote 88064 bytes at address 0x00015800 (0.26%)
Wrote 88064 bytes at address 0x0002b000 (0.39%)
Wrote 88064 bytes at address 0x00040800 (0.52%)
Wrote 88064 bytes at address 0x00056000 (0.66%)
Wrote 88064 bytes at address 0x0006b800 (0.79%)
Wrote 88064 bytes at address 0x00081000 (0.92%)
Wrote 88064 bytes at address 0x00096800 (1.05%)
Wrote 88064 bytes at address 0x000ac000 (1.18%)
Wrote 88064 bytes at address 0x000c1800 (1.31%)
Wrote 88064 bytes at address 0x000d7000 (1.44%)
Wrote 88064 bytes at address 0x000ec800 (1.57%)
Wrote 88064 bytes at address 0x00102000 (1.71%)
Wrote 88064 bytes at address 0x00117800 (1.84%)
Wrote 88064 bytes at address 0x0012d000 (1.97%)
Wrote 88064 bytes at address 0x00142800 (2.10%)
Wrote 88064 bytes at address 0x00158000 (2.23%)
Wrote 88064 bytes at address 0x0016d800 (2.36%)
Wrote 88064 bytes at address 0x00183000 (2.49%)
Wrote 88064 bytes at address 0x00198800 (2.62%)
Wrote 88064 bytes at address 0x001ae000 (2.76%)
Wrote 88064 bytes at address 0x001c3800 (2.89%)
Wrote 88064 bytes at address 0x001d9000 (3.02%)
Wrote 88064 bytes at address 0x001ee800 (3.15%)
Wrote 88064 bytes at address 0x00204000 (3.28%)
Wrote 88064 bytes at address 0x00219800 (3.41%)
Wrote 88064 bytes at address 0x0022f000 (3.54%)
Wrote 88064 bytes at address 0x00244800 (3.67%)
Wrote 88064 bytes at address 0x0025a000 (3.81%)
Wrote 88064 bytes at address 0x0026f800 (3.94%)
Wrote 88064 bytes at address 0x00285000 (4.07%)
Wrote 88064 bytes at address 0x0029a800 (4.20%)
Wrote 88064 bytes at address 0x002b0000 (4.33%)
Wrote 88064 bytes at address 0x002c5800 (4.46%)
Wrote 88064 bytes at address 0x002db000 (4.59%)
Wrote 88064 bytes at address 0x002f0800 (4.72%)
Wrote 88064 bytes at address 0x00306000 (4.86%)
Wrote 88064 bytes at address 0x0031b800 (4.99%)
Wrote 88064 bytes at address 0x00331000 (5.12%)

...

Complete: 0
```

**Comment by nornagon** [comment]
Thanks for reporting this and helping to make Electron better!

Because of time constraints, triaging code with third-party dependencies is usually not feasible for a small team like Electron's.

Would it be possible for you to make a standalone testcase with only the code necessary to reproduce the issue? For example, [Electron Fiddle](https://www.electronjs.org/fiddle) is a great tool for making small test cases and makes it easy to publish your test case to a [gist](https://gist.github.com) that Electron maintainers can use.

Stand-alone test cases make fixing issues go more smoothly: it ensure everyone's looking at the same issue, it removes all unnecessary variables from the equation, and it can also provide the basis for automated regression tests.

I'm adding the `blocked/need-repro` label for this reason. After you make a test case, please link to it in a followup comment.

Thanks in advance! Your help is appreciated.

**Comment by stephendpmurphy** [comment]
@nornagon I'm afraid creating a self contained test case will not be sufficient for getting to the root of this problem as the "third-party" .exe appears to be integral to the problem. Not the source or purpose of the .exe itself, but rather that it's a pre-compiled application in Windows. 

I believe this problem is specific to piping STDIO data from a pre-compiled binary via a child process in Electron (Again, the same source can be executed from Node directly and it works fine). Another data point is that I took the exact same source and executed it in Electron on a Linux platform. I replaced the pre-compiled .exe with the exact same version source but in a .bin format intended to run on Linux and the STDIO piping works as expected.

So it appears to be specific to piping STDIO from a pre-compiled binaries executed via a  child process on the Windows platform. 

I will gladly execute further tests on my system if you can give me some things to try.

**Comment by Prinzhorn** [comment]
> I'm afraid creating a self contained test case will not be sufficient for getting to the root of this problem as the "third-party" .exe appears to be integral to the problem

Does that mean using any other `.exe` that prints to stdout/stderr does not have the same problem? I'm not a Windows user, but there surely is something like `echo` you can use to test this?

Also self-contained test case does not necessarily mean it cannot use `sam-ba.exe`. It means code that can be copied and pasted and contains everything that is needed. E.g. your example results in

> Error: Cannot find module 'app-root-dir'

So if you can reproduce this with an `.exe` that is part of every Windows distribution you could share a self-contained test-case, preferable using a Fiddle gist.

**Comment by Prinzhorn** [comment]
> Just to reiterate I can execute the same source from Node directly and it works fine.

`electron` and `node` don't necessarily behave the same, but they should in most cases. One thing you can also test is if it behaves different when you run your code in a worker thread spawned from Electron/Node.

**Comment by stephendpmurphy** [comment]
@Prinzhorn thanks for the response. I just finished a test where I moved the source to a worker_thread and I get the same results. It works on everything EXCEPT when executing from Electron on WIndows.

Thanks for the feedback on the Fiddle/Self contained code. I will rework it and see what I can come up with. I am almost positive that the blanket statement of executing ANY .exe from Windows is not true ( I shouldn't have made that statement). I will try and find a couple .exes that ship with windows and see if I can find any that give similar behavior. My luck is that I will find NONE that can replicate it 😦
